src/messageComposer/messageComposer.ts

```diff
@@ -347,36 +352,30 @@ export class MessageComposer {
     });
   }
 
-  public registerSubscriptions = () => {
-    if (this.unsubscribeFunctions.size) {
+  public registerSubscriptions = (): UnregisterSubscriptions => {
+    if (this.hasSubscriptions) {
       // Already listening for events and changes
       return noop;
     }
-    this.unsubscribeFunctions.add(this.subscribeMessageComposerSetupStateChange());
-    this.unsubscribeFunctions.add(this.subscribeMessageUpdated());
-    this.unsubscribeFunctions.add(this.subscribeMessageDeleted());
-
-    this.unsubscribeFunctions.add(this.subscribeTextComposerStateChanged());
-    this.unsubscribeFunctions.add(this.subscribeAttachmentManagerStateChanged());
-    this.unsubscribeFunctions.add(this.subscribeLinkPreviewsManagerStateChanged());
-    this.unsubscribeFunctions.add(this.subscribePollComposerStateChanged());
-    this.unsubscribeFunctions.add(this.subscribeCustomDataManagerStateChanged());
-    this.unsubscribeFunctions.add(this.subscribeMessageComposerStateChanged());
-    this.unsubscribeFunctions.add(this.subscribeMessageComposerConfigStateChanged());
+    this.addUnsubscribeFunction(this.subscribeMessageComposerSetupStateChange());
+    this.addUnsubscribeFunction(this.subscribeMessageUpdated());
+    this.addUnsubscribeFunction(this.subscribeMessageDeleted());
+
+    this.addUnsubscribeFunction(this.subscribeTextComposerStateChanged());
+    this.addUnsubscribeFunction(this.subscribeAttachmentManagerStateChanged());
+    this.addUnsubscribeFunction(this.subscribeLinkPreviewsManagerStateChanged());
+    this.addUnsubscribeFunction(this.subscribePollComposerStateChanged());
+    this.addUnsubscribeFunction(this.subscribeCustomDataManagerStateChanged());
+    this.addUnsubscribeFunction(this.subscribeMessageComposerStateChanged());
+    this.addUnsubscribeFunction(this.subscribeMessageComposerConfigStateChanged());
     if (this.config.drafts.enabled) {
-      this.unsubscribeFunctions.add(this.subscribeDraftUpdated());
-      this.unsubscribeFunctions.add(this.subscribeDraftDeleted());
+      this.addUnsubscribeFunction(this.subscribeDraftUpdated());
```

Comment:
Would be great to have instant access to a specific unsubscribe function. Here for example, what if 

1. the drafts are disabled, 
2. enabled, 
3. disabled back

There are two issues:
1. The composer is not enabling the draft WS subscriptions on config change (1. -> 2.)
2. The composer is not unsubscribing the WS subscriptions on config change (1. -> 3.

The first issue can be solved immediately in the config state change subscriber, but for the second we do not have the setup that would allows us to take the specific unsubscribe function reference and call it.

Could we introduce a possibility to associate unsubscribe functions with identifiers or store direct references to them?

Reply:
Added proposals here: https://github.com/GetStream/stream-chat-js/pull/1529

Reply:
Copied suggestions over from #1529 in  [900178b](https://github.com/GetStream/stream-chat-js/pull/1522/commits/900178b8d1beabfe88f9f6a77527cb63c093cfaf).